### DISCUSSION TITLE
How to change indentation, in shell script

### DISCUSSION THREAD

**Discussion by nianfouyi** [LazyVim/LazyVim#1247] [Q&A]
Hello everyone, I use lazyvim to write code, but when I use it to write shell code, its indentation is 2 spaces, unfortunately, I will use vscode to write the same file at the same time, the indentation set in vscode is four Spaces, how should I set to use two spaces as indentation in shell scripts, if someone can point me, I will be grateful, I use the latest lazyvim and ubuntu 22.04.

**Comment by dpetka2001** [comment]
`~/.config/nvim/after/ftplugin/sh.lua` (assuming that the file you want to edit is of filetype `sh` as are most bash scripts, but you should check with `set filetype?` just to be sure and change the filename accordingly)
```shell
vim.opt_local.tabstop = 4
vim.opt_local.shiftwidth = 4
vim.opt_local.expandtab = true
```
Change the numbers to what you like.

**Reply by pkazmier** [reply]
I’m curious, is the above the preferred way of setting file-specific options? I recently had to make a tweak for Go files, but added an autocmd to my `config/autocmds.lua` instead. Is one way preferred over another?

```lua
vim.api.nvim_create_autocmd("FileType", {
  pattern = { "go" },
  callback = function()
    vim.opt_local.tabstop = 8
    vim.opt_local.shiftwidth = 8
    vim.opt_local.expandtab = false
  end,
})
```

**Reply by dpetka2001** [reply]
I think it's just a matter of when the files are loaded. `after` directory is processed before Lazyvim's autocmd if I'm not mistaken. As to what to prefer, I'm not too sure myself. I'm just used to this from when I was using vim itself and just prefered to set the options in a file in the `after/ftplugin` directory instead of doing an autocmd.

**Reply by nianfouyi** [reply]
Unfortunately, I did not find the after folder under ~/.config/nvim, do I need to create it?

**Reply by dpetka2001** [reply]
Yes, you need to create it yourself. Create both folders. `after/plugin/` should be both directories and you put the files with the extensions you want inside.
